colossalai/auto_parallel/meta_profiler/metainfo.py

```diff
@@ -80,9 +81,10 @@ def compute_sharded_tensor(self, operation_data: OperationData, sharding_spec: S
             else:
                 # sharded according to device_mesh shape
                 new_shape.append(dim // np.prod(np.array([device_mesh.mesh_shape[i] for i in shard.shard_list])))
-
+        new_data = torch.zeros(new_shape, device="meta")
+        new_data.data_ptr = operation_data.data.data_ptr
         return OperationData(name=operation_data.name,
-                             data=torch.zeros(new_shape, device="meta"),
+                             data=new_data,
```

Comment:
Each new data fed into `OperationData` should be marked with `data_ptr`

Reply:
What if there are two tensors which share the same underlying data?